src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/entity/BookingHistory.kt

```diff
@@ -0,0 +1,87 @@
+package uk.gov.justice.digital.hmpps.hmppsbookavideolinkapi.entity
+
+import jakarta.persistence.CascadeType
+import jakarta.persistence.Entity
+import jakarta.persistence.FetchType
+import jakarta.persistence.GeneratedValue
+import jakarta.persistence.GenerationType
+import jakarta.persistence.Id
+import jakarta.persistence.JoinColumn
+import jakarta.persistence.ManyToOne
+import jakarta.persistence.OneToMany
+import jakarta.persistence.Table
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.LocalTime
+
+/**
+ * This entity represents the history of a booking at a point in time, and is created/saved
+ * whenever a video link booking is CREATED, AMENDED or CANCELLED. Upon each the details of the
+ * related appointments should also be recorded as a list.
+ */
+
+@Entity
+@Table(name = "booking_history")
+class BookingHistory(
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  val bookingHistoryId: Long = 0L,
+
+  val videoBookingId: Long,
+
+  val historyType: String,
+
+  val courtId: Long? = null,
+
+  val hearingType: String? = null,
+
+  val probationTeamId: Long? = null,
+
+  val probationMeetingType: String? = null,
+
+  val videoUrl: String? = null,
+
+  val comments: String? = null,
+
+  val createdBy: String,
+
+  val createdTime: LocalDateTime = LocalDateTime.now(),
+
+  @OneToMany(mappedBy = "bookingHistory", fetch = FetchType.EAGER, cascade = [CascadeType.ALL], orphanRemoval = true)
+  val appointments: MutableList<BookingHistoryAppointment> = mutableListOf(),
```

Comment:
I would suggest taking `appointments` out of the primary constructor (and making it private) given there is the `addBookingHistoryAppointments` method below.  Perhaps look at `VideoBooking` entity for an example.

As it stands this can be mutated directly e.g. `history.appointments.add` ....

Reply:
Good call. Will do.